### PR TITLE
feat: Fix cost calculations for AWS S3

### PR DESCRIPTION
**Pull request by bishal7679**
## Problem #1022

Currently cost for AWS S3 is always showing as $0.
 
## Solution

This PR is fixing that issue by adding more filters and usage metrics for request to s3 bucket with absolute pricemap fetching.

## Changes Made

- Fetching price info using AWS SDK price module using `AmazonS3` service code and appropriate filters.
- getting price map using util function GetPriceMap from [utils.go](https://github.com/tailwarden/komiser/blob/develop/providers/aws/utils/utils.go)
- fetching resource objects using S3 client
- also fetching usage metrics as request using cloudwatch
- Finally added calculation as per all the metrics

## Checklist

- [x] Code follows the <a href="https://github.com/tailwarden/komiser/blob/master/CONTRIBUTING.md">contributing</a> guidelines
- [x] Changes have been thoroughly tested
- [ ] <a href="https://github.com/tailwarden/docs.komiser.io">Documentation</a> has been updated, if necessary
- [ ] Any dependencies have been added to the project, if necessary

### CODE REVIEW

File: providers/aws/s3/buckets.go [hunk 1]
```diff
@@ -9,69 +9,143 @@ import (
...
+				Type:  types.FilterTypeTermMatch,
+			},
+			{
+				Field: aws.String("productFamily"),
+				Value: aws.String("Storage"),
+				Type:  types.FilterTypeTermMatch,
+			},
+			{
+				Field: aws.String("storageClass"), // Filter by storage class
+				Value: aws.String("STANDARD"),     // Specify the desired storage class
```
Comment (AvineshTripathi) [inline providers/aws/s3/buckets.go:48]: Lets make this dynamic

File: providers/aws/s3/buckets.go [hunk 2]
```diff
@@ -9,69 +9,143 @@ import (
...
 
 		if sizeInTB <= 50 {
-			monthlyCost = (sizeInTB * 1000) * 0.023
+			storageCostPerGB = (sizeInTB * 1024) * 0.023
 		} else if sizeInTB <= 450 {
-			monthlyCost = (sizeInTB * 1000) * 0.022
+			storageCostPerGB = (sizeInTB * 1024) * 0.022
 		} else {
-			monthlyCost = (sizeInTB * 1000) * 0.021
+			storageCostPerGB = (sizeInTB * 1024) * 0.021
```
Comment (Azanul) [inline providers/aws/s3/buckets.go:109]: Replace hardcoded values with actual values from pricing output similar to as you've done for requests

File: providers/aws/s3/buckets.go [hunk 3]
```diff
@@ -9,69 +9,143 @@ import (
...
+			{
+				Field: aws.String("productFamily"),
+				Value: aws.String("Storage"),
+				Type:  types.FilterTypeTermMatch,
+			},
+			{
+				Field: aws.String("storageClass"), // Filter by storage class
+				Value: aws.String("STANDARD"),     // Specify the desired storage class
+				Type:  types.FilterTypeTermMatch,
+			},
```
Comment (Azanul) [inline providers/aws/s3/buckets.go:50]: different buckets might have different storage classes

**Review by Azanul — CHANGES_REQUESTED** [review]
You've pushed 2 unwanted files: azure.svg & favicon.ico

File: providers/aws/s3/buckets.go (hunk 1 above)
Comment (bishal7679) [reply providers/aws/s3/buckets.go:48]: ok :+1:

**Comment by bishal7679** [comment]
> You've pushed 2 unwanted files: azure.svg & favicon.ico

i dont understand! it's been taken automatically :(

**Comment by Azanul** [comment]
> i dont understand! it's been taken automatically :(

commit only the files you want to commit, son't commit all

File: providers/aws/s3/buckets.go (hunk 2 above)
Comment (bishal7679) [reply providers/aws/s3/buckets.go:109]: what's wrong with it as it covering 500 TB. When we are trying to fetch it using GetCost then it is giving $0 i.g its eliminating all the so small charges.

Comment (bishal7679) [reply providers/aws/s3/buckets.go:109]: i can do it but you will unable to see so small amount like $0.00000011

Comment (Azanul) [reply providers/aws/s3/buckets.go:109]: I think it's taking free tier into account, if the free tier is described by the pricing output.

Comment (bishal7679) [reply providers/aws/s3/buckets.go:109]: could you pls check once with my code by removing the static values and adding `awsUtils.GetCost(priceMap["AWS-S3-Storage"]` dynamically with your AWS account??
and taking `sizeinTB` into account

**Pull request closed without merging**